prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c

```diff
@@ -135,8 +133,11 @@ ssize_t fi_ibv_rdm_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 						    threshold - counter,
 						    src_addr);
 		counter += (ret > 0) ? ret : 0;
-	} while ((ret >= 0) && (counter < threshold ||
-		(timeout >= 0 && fi_gettime_ms() < time_limit)));
```

Comment:
We can simplify this if we check for an infinite timeout (-1) up front, and set time_limit = SIZE_MAX above.  I think that's the intent of this patch.  This should allow:
```
while ((ret >= 0) && (counter < threshold) && (fi_gettime_ms() < time_limit))
```
Why is the logic OR between the counter threshold and time checks